### ISSUE TITLE
Can't find values from environments.yaml under bases

### ISSUE THREAD

**Issue by tstmrk** [roboll/helmfile#1142]
Probably I am doing something wrong, but I can't see why. I constantly get the error
```
in ./helmfile.yaml: error during helmfile.yaml.part.0 parsing: template: stringTemplate:7:20: executing "stringTemplate" at <.Values.nginx.releaseName>: map has no entry for key "nginx"
``` 

**The situation**

helmfile.yaml
```
bases:
  - environments.yaml

releases:
  - name: {{ .Values.nginx.releaseName }}
    chart: bitnami/nginx
```
environments.yaml
```
environments:
  default:
    values:
      - nginx:
          releaseName: nginx
```

Can someone help me out?

Version info
helmfile version v0.100.0

**Comment by mumoshu** [comment]
I think you have a chicken-and-egg problem between the environment values loaded by parsing yaml but to render the yaml you need environment values get loaded.

Try separating environment values definitions and your releases with `---`:

```
bases:
  - environments.yaml
---
releases:
  - name: {{ .Values.nginx.releaseName }}
    chart: bitnami/nginx
```

**Comment by tstmrk** [comment]
> I think you have a chicken-and-egg problem between the environment values loaded by parsing yaml but to render the yaml you need environment values get loaded.
> 
> Try separating environment values definitions and your releases with `---`:
> 
> ```
> bases:
>   - environments.yaml
> ---
> releases:
>   - name: {{ .Values.nginx.releaseName }}
>     chart: bitnami/nginx
> ```

Unfortunately this doesn't work either. Do you have an example in which setup it will work? Even with below setup I can't get it working.

helmfile.yaml
```
helmfiles:
  - nginx.yaml
```
nginx.yaml
```
bases:
  - environments.yaml
---
releases:
  - name: nginx
    chart: bitnami/nginx
    values:
      - hosts:
        - name: {{ .Values.nginx.host }}
          path: /
```
environments.yaml
```
environments:
  default:
    values:
      - nginx:
          host: something.local
```

I will get the error message:
```
root@54899d99778a:~/workspace/helm/github-1142# helmfile -f helmfile-releases.yaml build
in ./helmfile-releases.yaml: in .helmfiles[0]: in ./nginx.yaml: error during nginx.yaml.part.0 parsing: template: stringTemplate:9:26: executing "stringTemplate" at <.Values.nginx.host>: map has no entry for key "nginx"
```

**Comment by yarinm** [comment]
@tstmrk change your environments.yaml to the following:
```
environments:
  default:
    values:
      nginx:
          host: something.local
```

When environments.yaml is included for some reason values are passed as a map

**Comment by tstmrk** [comment]
> @tstmrk change your environments.yaml to the following:
> 
> ```
> environments:
>   default:
>     values:
>       nginx:
>         host: something.local
> ```
> 
> When environments.yaml is included for some reason values are passed as a map

Thanks for your suggestion, but this didn't work out either. I am still getting the same error. I tried a couple of situations:

* Seperate helm files
* with and without the YAML-file seperator  `---`
* values as a map, and values as an array
* unnested map/nested map (`.Values.nginx.host / .Values.host `)
* Use `.Environment.Values.nginx.host` instead of `.Values.nginx.host`

None of the above solves the issue.

**Comment by mumoshu** [comment]
@tstmrk Hey! How's your `helmfile-releases.yaml` seen in your example look like?

I'm asking because I couldn't reproduce the issue i.e. the below works on my machine:

helmfile.yaml

```
helmfiles:
  - nginx.yaml
```

nginx.yaml

```
bases:
  - environments.yaml
---
releases:
  - name: nginx
    chart: bitnami/nginx
    values:
      - hosts:
        - name: {{ .Values.nginx.host }}
          path: /
```

environments.yaml

```
environments:
  default:
    values:
      - nginx:
          host: something.local
```

`helmfile build`:

```
$ helmfile build
---
#  Source: nginx.yaml

filepath: nginx.yaml
environments:
  default:
    values:
    - nginx:
        host: something.local
bases:
- environments.yaml
releases:
- chart: bitnami/nginx
  name: nginx
  values:
  - hosts:
    - name: something.local
      path: /
templates: {}
---
#  Source: helmfile.yaml

filepath: helmfile.yaml
helmfiles:
- path: nginx.yaml
  environment: {}
templates: {}
```

**Comment by tstmrk** [comment]
@mumoshu, thanks for the reply. I made an exact copy of your examples, but got exactly the same error. 

```
root@54899d99778a:~/workspace/helm/github-1142/mumoshu# ls -la
total 2
drwxrwxrwx    2 root     root             0 Mar 17 11:54 .
drwxrwxrwx    2 root     root             0 Mar 17 11:54 ..
-rwxr-xr-x    1 root     root            87 Mar 17 12:13 environments.yaml
-rwxr-xr-x    1 root     root            26 Mar 17 12:12 helmfile.yaml
-rwxr-xr-x    1 root     root           178 Mar 17 12:13 nginx.yaml
```

Result
```
root@54899d99778a:~/workspace/helm/github-1142/mumoshu# helmfile build
in ./helmfile.yaml: in .helmfiles[0]: in ./nginx.yaml: error during nginx.yaml.part.0 parsing: template: stringTemplate:9:26: executing "stringTemplate" at <.Values.nginx.host>: map has no entry for key "nginx"
```

When I change `{{ .Values.nginx.host }}` nginx.yaml into `something.local`, it doesn't fail. But that's not what I want :-)

```
root@54899d99778a:~/workspace/helm/github-1142/mumoshu# helmfile build
---
#  Source: nginx.yaml

filepath: nginx.yaml
environments:
  default:
    values:
    - nginx:
        host: something.local
bases:
- environments.yaml
releases:
- chart: bitnami/nginx
  name: nginx
  values:
  - hosts:
    - name: something.local
      path: /
templates: {}
---
#  Source: helmfile.yaml

filepath: helmfile.yaml
helmfiles:
- path: nginx.yaml
  environment: {}
templates: {}
```

**Helm version**
```version.BuildInfo{Version:"v3.1.0", GitCommit:"b29d20baf09943e134c2fa5e1e1cab3bf93315fa", GitTreeState:"clean", GoVersion:"go1.13.7"}```

```
root@54899d99778a:~/workspace/helm/github-1142/mumoshu# uname -a
Linux 54899d99778a 4.9.125-linuxkit #1 SMP Fri Sep 7 08:20:28 UTC 2018 x86_64 Linux
```

Do you need some extra information?

**Comment by tstmrk** [comment]
Aahh!! I have got the cause! 

Regarding my example it turned out that
* the _End of Line Sequence_ of the file `nginx.yaml` must be _LF_ and not _CRLF_ and 
* there must be a YAML-seperator (`---`) in the file.

I am developing my helm chart on a Windows PC in VSCode and my default End of Line Sequence is _CRLF_. I am testing it inside a docker container, which only supports _LF_

The only strange thing left is that I tested exactly the same situation on my Windows PC (nginx.yaml with eol sequence = CRLF) and it failed also. So, even if you're testing on a Windows PC the eol sequence must be LF?
